posthog/models/test/test_person_override_model.py

```diff
@@ -2,9 +2,10 @@
 from uuid import uuid4
 
 import pytest
+from django.db import connection
 from django.db.utils import IntegrityError
 
-from posthog.models import PersonOverride, Team
+from posthog.models import PersonOverride, PersonOverrideHelper, Team
```

Comment:
Could you add the extra changes/tests from the reverted PR https://github.com/PostHog/posthog/pull/14277
- use different team/org
```
organization = create_organization(name="test")
team = create_team(organization=organization)
```
instead of 
```
PersonOverride.objects.all().delete()
PersonOverrideHelper.objects.all().delete()
```
- test_person_override_disallows_overriding_to_itself - Test old_person_id cannot match override_person_id
- test_person_override_disallows_concurrent_merge

Reply:
`create_team` and `create_organization` do not delete state after test is finished. I would use the fixtures in #14513

Reply:
and the fact that they don't is great - makes debugging much easier as you can inspect the db state after the test

Reply:
But that creates inconsistent state for each test.

Reply:
I've brought over the new tests from #14227 (albeit in their updated form as seen in #14513), as well as the comment edit pointed out in #14513.

Locally they are passing, but waiting for CI to be green too.